pkg/kubelet/stats/cadvisor_stats_provider.go

```diff
@@ -303,20 +300,58 @@ func isPodManagedContainer(cinfo *cadvisorapiv2.ContainerInfo) bool {
 
 // getCadvisorPodInfoFromPodUID returns a pod cgroup information by matching the podUID with its CgroupName identifier base name
 func getCadvisorPodInfoFromPodUID(podUID types.UID, infos map[string]cadvisorapiv2.ContainerInfo) *cadvisorapiv2.ContainerInfo {
-	for key, info := range infos {
+	if info, found := infos[cm.GetPodCgroupNameSuffix(podUID)]; found {
+		return &info
+	}
+	return nil
+}
+
+// filterTerminatedContainerInfoAndAssembleByPodCgroupKey returns the specified containerInfo but with
+// the stats of the terminated containers removed and all containerInfos assembled by pod cgroup key.
+//
+// A ContainerInfo is considered to be of a terminated container if it has an
+// older CreationTime and zero CPU instantaneous and memory RSS usage.
+func filterTerminatedContainerInfoAndAssembleByPodCgroupKey(containerInfo map[string]cadvisorapiv2.ContainerInfo) (map[string]cadvisorapiv2.ContainerInfo, map[string]cadvisorapiv2.ContainerInfo) {
```

Comment:
I think this replaces `removeTerminatedContainerInfo` now and I don't see any more usages of `removeTerminatedContainerInfo`. Can we delete it then?